packages/next/src/builders/build/lib/create-package-json.ts

```diff
@@ -2,13 +2,41 @@ import { BuilderContext } from '@angular-devkit/architect';
 import { writeFileSync } from 'fs';
 import { join } from 'path';
 import { readJsonFile } from '@nrwl/workspace';
+import { createProjectGraph } from '@nrwl/workspace/src/core/project-graph';
 import { NextBuildBuilderOptions } from '../../../utils/types';
 import { reactDomVersion, reactVersion } from '@nrwl/react/src/utils/versions';
 
+interface PackageJSONDeps {
+  dependencies: { [key: string]: string };
+  devDependencies: { [key: string]: string };
+  peerDependencies?: { [key: string]: string };
+  [key: string]: unknown;
+}
+
 export async function createPackageJson(
   options: NextBuildBuilderOptions,
   context: BuilderContext
 ) {
+  const graph = createProjectGraph();
+  const projectDependencies = graph.dependencies[context.target.project].map(
+    (dep) => dep.target
+  );
+  const projectNpmDependencies = Object.values(graph.nodes).reduce(
+    (acc, node) => {
+      if (node.type === 'npm' && projectDependencies.includes(node.name)) {
+        if (node.data.isDevDependency) {
```

Comment:
I think we can ignore dev and peer dependencies. Peer shouldn't be used for apps so we should just ignore.

Dev dependencies should all go into the root `package.json` so I think they can be ignored as well. Unless there's something I am missing.